### PR TITLE
Fix metrics in ack mgr

### PR DESCRIPTION
**Pull request by georgeteo**
- [bug] a couple counters fail to increment.
- add metrics for ack and nack errors.
- propagate ack/nack errors to user so that they can decide what to do. We currently give a false impression that an ack was successful even if there was an error in the list.

### CODE REVIEW

File: internal/consumer/ackMgr.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func newAckManager(maxOutstanding int, scope tally.Scope, logger *zap.Logger) *a
 func (mgr *ackManager) GetAckID(msgSeq int64) (ackID, error) {
 	addr, err := mgr.unackedSeqList.Add(msgSeq)
 	if err != nil {
-		mgr.tally.Counter(metrics.KafkaPartitionGetAckIDErrors)
+		mgr.tally.Counter(metrics.KafkaPartitionGetAckIDErrors).Inc(1)
```
Comment (venkat1109) [inline internal/consumer/ackMgr.go]: wow

File: internal/consumer/ackMgr.go [hunk 2]
```diff
@@ -110,20 +110,27 @@ func (mgr *ackManager) GetAckID(msgSeq int64) (ackID, error) {
 }
 
 // Ack marks the given msgSeqNum as processed
-func (mgr *ackManager) Ack(id ackID) {
-	err := mgr.unackedSeqList.Remove(id.listAddr, id.msgSeq)
```
[the remark this answers is not in the document]
Comment (venkat1109) [reply internal/consumer/ackMgr.go]: I believe the only reason for Remove() to fail was a duplicate Ack() and the current offset is already moved forward. In that case, returning error wasn't going to help much for the caller. So, that's probably why it was ignored earlier.

If you do return an error from this method, you should think about what the caller is supposed to do (and also about the possible cases when error will be returned)

File: internal/consumer/ackMgr.go [hunk 3]
```diff
@@ -110,20 +110,27 @@ func (mgr *ackManager) GetAckID(msgSeq int64) (ackID, error) {
...
 }
 
 // Nack marks the given msgSeqNum as processed, the expectation
 // is for the caller to move the message to an error queue
 // before calling this
-func (mgr *ackManager) Nack(id ackID) {
-	mgr.Ack(id)
-	mgr.tally.Counter(metrics.KafkaPartitionNacks)
+func (mgr *ackManager) Nack(id ackID) (err error) {
+	err = mgr.unackedSeqList.Remove(id.listAddr, id.msgSeq)
```
Comment (venkat1109) [inline internal/consumer/ackMgr.go]: same comment as above. Document what the caller is supposed to do. In the case of a duplicate ack or nack, we don't want the caller to sit in a tight loop and retry - it won't help with anything

**Comment by georgeteo** [comment]
I decided to eat the errors since they're not user actionable. We can revisit how do handle these errors better later. I'd still like to get this PR in for the metrics fixes. :)

**Pull request merged**